store-implementation/federated-store/src/main/java/uk/gov/gchq/gaffer/federatedstore/operation/handler/impl/FederatedOperationHandler.java

```diff
@@ -140,4 +161,31 @@ private List<GraphSerialisable> getGraphs(final FederatedOperation<INPUT, OUTPUT
                 graphs
                 : Collections.emptyList();
     }
+
+    /**
+     * Checks if the schemas of the supplied graphs have any of the same groups.
+     *
+     * @param graphs The list of graphs.
+     * @return True if any schema groups are shared.
+     */
+    private boolean graphsHaveCommonSchemaGroups(final List<GraphSerialisable> graphs) {
+        // Check if any of the graphs have common groups in their schemas
+        List<Schema> graphSchemas = graphs.stream()
+            .map(GraphSerialisable::getSchema)
+            .collect(Collectors.toList());
+
+        // Compare all schemas against each other
+        for (int i = 0; i < graphSchemas.size() - 1; i++) {
```

Comment:
Might be nicer to use [cartesianProduct](https://guava.dev/releases/snapshot/api/docs/com/google/common/collect/Lists.html#cartesianProduct(java.util.List...))

Reply:
I was trying not to rely on an external library with cartesian product I'd still have to do a double loop as you'd end up with a list of lists that have just have two elements.

Reply:
I was thinking `for pair in cartesianProduct(schemas, schemas)`, but you would have to check the schemas aren't the same so a tad awkward

Reply:
Yeah I'm not sure it would change too much of the logic it would end up something like:
```java
List<List<Schema>> product = Lists.cartesianProduct(graphSchemas, graphSchemas);

for (List<Schema> pair : product) {
    Set<String> firstGroupSet = pair.get(0).getGroups();
    Set<String> secondGroupSet = pair.get(1).getGroups();
    if (firstGroupSet.stream().anyMatch(secondGroupSet::contains)) {
        LOGGER.debug("Found common schema groups between requested graphs");
        return true;
    }
}
```
I'd rather not use an external library if its not required

Reply:
I'd agree regarding the use of external libraries. I've added a separate suggestion regarding the loop.